spec/spec_helper.rb

```diff
@@ -57,37 +62,37 @@
     Hermes.origin_headers = {}
   end
 
-  database_name = "hermes-rb-test"
-  ENV["DISTRIBUTED_TRACING_DATABASE_URI"] ||= "postgresql://localhost/#{database_name}"
+  ENV["DATABASE_URL"] ||= "postgresql://localhost/hermes-rb-test"
+  ENV["DISTRIBUTED_TRACING_DATABASE_URI"] ||= ENV["DATABASE_URI"]
 
-  ActiveRecord::Base.establish_connection(adapter: "postgresql", database: database_name)
-  begin
-    database = ActiveRecord::Base.connection
-  rescue ActiveRecord::NoDatabaseError
-    ActiveRecord::Base.establish_connection(adapter: "postgresql").connection.create_database(database_name)
-    ActiveRecord::Base.establish_connection(adapter: "postgresql", database: database_name)
-    database = ActiveRecord::Base.connection
+  if Gem::Version.new(ActiveRecord::VERSION::STRING) > Gem::Version.new('6')
+    # Rails 5.2 is unable to handle this
+    ActiveRecord::Tasks::DatabaseTasks.create(ENV["DATABASE_URL"])
   end
 
-  database.drop_table(:hermes_distributed_traces) if database.table_exists?(:hermes_distributed_traces)
-  database.create_table(:hermes_distributed_traces) do |t|
-    t.string "trace", null: false
-    t.string "span", null: false
-    t.string "parent_span"
-    t.string "service", null: false
-    t.text "event_class", null: false
-    t.text "routing_key", null: false
-    t.jsonb "event_body", null: false, default: []
-    t.jsonb "event_headers", null: false, default: []
-    t.datetime "created_at", precision: 6, null: false
-    t.datetime "updated_at", precision: 6, null: false
-
-    t.index ["created_at"], name: "index_hermes_distributed_traces_on_created_at", using: :brin
-    t.index ["trace"], name: "index_hermes_distributed_traces_on_trace"
-    t.index ["span"], name: "index_hermes_distributed_traces_on_span"
-    t.index ["service"], name: "index_hermes_distributed_traces_on_service"
-    t.index ["event_class"], name: "index_hermes_distributed_traces_on_event_class"
-    t.index ["routing_key"], name: "index_hermes_distributed_traces_on_routing_key"
+  ActiveRecord::Base.establish_connection(ENV["DATABASE_URL"])
```

Comment:
so this DB setup works just like that? 

Seems like GH actions allows for a way better setup, as far as I remember, this wasn't the case on CircleCI and it required some very frustrating hacks here.

Reply:
No, actually, I also create the DB if it doesn't exist. I just use the rails default approach on this - oneliner above.
But this now requires only for local setup simplification. On GHA, we are using the official PG docker image which has the default DB created on start and I provide this default name as an ENV variable. On CircleCI it is also possible since it uses docker for services as well. But Travis - I don't think so.